ospfd/ospfd.c

```diff
@@ -398,9 +398,6 @@ struct ospf *ospf_get(unsigned short instance, const char *name, bool *created)
 		ospf = ospf_new(instance, name);
 		ospf_add(ospf);
 
-		if (ospf->router_id_static.s_addr == INADDR_ANY)
```

Comment:
How is this dead code?  Isn't it possible for ospf to be coming up without a router id?

Reply:
The dead code is the next step - in the `ospf_router_id_update` function. It is described in the commit: `ospf_router_id_update` checks for `ospf->oi_running`, but it'll never be 1 right after creation by ospf_new.